### PR TITLE
Add a rule to disallow specific imports if flow isn't being used

### PR DESCRIPTION
**Pull request by kevinbarabash**
The new `imports-requiring-flow` rule allows you to specify a which imports (either npm packages or local files) will require using `// @flow` in the file doing the import/require.  In order for this to work, you have to specify a `rootDir` in the rule configuration.

Test Plan:
- `yarn link`
- in devtools/khan-linter: `yarn link "eslint-plugin-khan"
- add an eslint.custom.js config to webapp containing the code listed below
- modify .eslintrc in webapp to include eslint.custom.js
- check that `404-page.jsx` and `foo` can't be imported

```js
// This has to be a .js file since we need to use __dirname
module.exports = {
    plugins: ["khan"],
    rules: {
        "khan/imports-requiring-flow": [
            "error",
            {
                rootDir: __dirname,
                modules: ["foo", "javascript/404-package/404-page.jsx"],
            },
        ],
    },
};
```

Screen Shot 2019-10-23 at 12 04 31 PM
Screen Shot 2019-10-22 at 5 03 41 PM

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    rules: {
+        "flow-array-type-style": require("./rules/flow-array-style.js"),
+        "flow-no-one-tuple": require("./rules/flow-no-one-tuple.js"),
+        "imports-requiring-flow": require("./rules/imports-requiring-flow.js"),
+        "react-no-subscriptions-before-mount": require("./rules/react-no-subscriptions-before-mount.js"),
+    },
```
Comment (kevinbarabash) [inline lib/index.js]: It took me a while to figure out that we needed to put all of the rules inside a `rules` property.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "eslint-plugin-khan",
-  "version": "0.0.1",
-  "main": "index.js",
+  "name": "@khanacademy/eslint-plugin-khan",
```
Comment (kevinbarabash) [inline package.json]: We want to namespace all of our packages right?  In this case it's a little redundant though.

File: lib/rules/imports-requiring-flow.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        let usingFlow = false;
+
+        return {
+            Program(node) {
+                usingFlow = node.comments.some(
+                    comment => comment.value.trim() === "@flow",
+                );
+            },
+            ImportDeclaration(node) {
```
Comment (jeresig) [inline lib/rules/imports-requiring-flow.js]: So if I'm reading this correctly, we're not handling dynamic import calls, right? It feels like we'll want to handle that, as well!

**Comment by kevinbarabash** [comment]
Good call on handling dynamic imports.  That shouldn't be too hard to add.

**Pull request merged**